src/main/java/com/revature/studyforce/flashcard/service/FlashcardService.java

```diff
@@ -0,0 +1,224 @@
+package com.revature.studyforce.flashcard.service;
+
+import com.revature.studyforce.flashcard.dto.FlashcardAllDTO;
+import com.revature.studyforce.flashcard.dto.FlashcardDTO;
+import com.revature.studyforce.flashcard.dto.NewFlashcardDTO;
+import com.revature.studyforce.flashcard.model.Difficulty;
+import com.revature.studyforce.flashcard.model.Flashcard;
+import com.revature.studyforce.flashcard.model.Topic;
+import com.revature.studyforce.flashcard.repository.FlashcardRepository;
+import com.revature.studyforce.flashcard.repository.TopicRepository;
+import com.revature.studyforce.user.model.User;
+import com.revature.studyforce.user.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Sort;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+import java.sql.Timestamp;
+import java.time.LocalDateTime;
+import java.util.Locale;
+import java.util.Optional;
+
+/**
+ * Services for the Flashcard repository
+ * @author Luke
+ */
+@Service
+public class FlashcardService implements AbstractService {
+
+    private final FlashcardRepository flashcardRepository;
+    private final TopicRepository topicRepository;
+    private final UserRepository userRepository;
+
+    @Autowired
+    public FlashcardService(TopicRepository topicRepository, FlashcardRepository flashcardRepository, UserRepository userRepository){
+        this.flashcardRepository =flashcardRepository;
+        this.userRepository = userRepository;
+        this.topicRepository = topicRepository;
+    }
+
+    /**
+     * getAll() method mapped to HTTP GET requests ("/all")
+     * @param page - number of offsets away from 0
+     * @param offset - number of Flashcards per page
+     * @param sortBy - column to sort by
+     * @param order - ascending or descending order
+     * @return - returns Page of paginated Flashcards
+     */
+    public Page<FlashcardAllDTO> getAll(int page, int offset, String sortBy, String order){
```

Comment:
You are validating input in almost all your methods. So your documentation needs to reflect the constraints of the parameters.